### PR TITLE
Replace links with Jetpack redirect service for migration plugin

### LINKED ISSUE #29128: Update Learn more link to the new documentation page

### PR DESCRIPTION
**Pull request by ouikhuan**
Fixes #29128 

## Proposed changes:

* Replace all the static links in the plugin with the Jetpack redirect manager.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No
## Testing instructions:

* Spin up a [JN site](https://jurassic.ninja/create/?jetpack-beta&branches.wpcom-migration=update/migration-doc-link&wp-debug-log) and navigate to Move to WordPress.com
* Click on the Learn more link and see if it brings you to `https://wordpress.com/support/import-using-wordpress-migration/`
* The doc is still in draft now and will be live on March 1st
* Click on `Contact us`, `Get Started`, `Check your migration progress` buttons to see if they get linked correctly like it was.

### CODE REVIEW

**Review by bogiii — DISMISSED** [review]
👍

File: projects/plugins/migration/src/js/components/migration/index.tsx [hunk 1]
```diff
@@ -130,7 +130,7 @@ export function Migration( props: Props ) {
 					isSecondary={ true }
 					target={ '_blank' }
 					href={ getRedirectUrl(
-						'https://wordpress.com/support/import/import-an-entire-wordpress-site/'
+						'https://wordpress.com/support/import-using-wordpress-migration/'
```
Comment (jeherve) [inline projects/plugins/migration/src/js/components/migration/index.tsx]: I would recommend using a Jetpack Redirect unique slug in here, so you can update that URL without having to release the plugin later on. You can read more about how to do that here:
PCYsg-pY7-p2

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/plugins/migration/src/js/components/migration/index.tsx (hunk 1 above)
Comment (ouikhuan) [reply projects/plugins/migration/src/js/components/migration/index.tsx]: Thanks for the suggestion @jeherve ! In this case I've updated all the static link in the plugin with the slug 👍

**Review by bogiii — APPROVED** [review]
Nice feature! Didn't know about this mechanism.

Learnt something new. Thanks! 🚀

**Review by jeherve — APPROVED** [review]

**Pull request merged**